use docker info or default gateway to guess host address

hkaj opened the pull request:
in some setups, using the hostname from `docker info` doesn't work resolve to the host internal IP, so we fallback to the default gateway in this case.

remh commented on alpine/entrypoint.sh:
```diff
@@ -103,8 +113,17 @@ fi
+    server=$(get_hostname)
+
+    # check if it resolves to the mesos slave
+    curl http://$server:5051/state.json
```
might want a timeout here ?

remh replied:
so basically 
```curl -I -s -m 1 http://$server:5051/state.json > /dev/null```

remh commented on entrypoint.sh:
```diff
@@ -104,8 +114,16 @@ fi
+    server=$(get_hostname)
+
+    # check if it resolves to the mesos slave
+    curl http://$server:5051/state.json
```
Same here

hkaj commented:
thanks for the review @remh, addressed your comments.

remh commented:
172.17.42.1 is not the standard anymore. See https://github.com/DataDog/docker-dd-agent/issues/114 for details.

We should discover the default gateway instead

hkaj commented:
@remh I removed `172.17.42.1` and added the default gateway back. Hopefully we won't need these two though as `http://169.254.169.254/latest/meta-data/local-ipv4` provides the right address.

remh commented:
@hkaj `http://169.254.169.254/latest/meta-data/local-ipv4` is an AWS specific API

hkaj commented:
:facepalm: I knew this felt too familiar... Removed it.
